### PR TITLE
FFT cleanup

### LINKED ISSUE #127: FFT implementation
/src/aspire/utils/fft.py uses scipy FFT and not FFTW which is supposed to be much faster.
What is the reason for that?

### PR DESCRIPTION
**Pull request by garrettwrong**
This purges some dead FFT code and removes some remaining FFT calls not using `aspire.numeric.fft`.

Some calls used padding semantics specific to `pocketfft` which are now explicit in the code, so we can use the generic fft calls.  If we use that a lot, we can migrate to the wrappers.

Closes #127 

~Noting that some of the `aspire.numeric.fft` facade calls to `PyFFTW` are currently (were) broken.
I've change the config to `scipy` so I can see the cross-platform CI results. I will change back to PyFFTW after resolving that bug. Working in it now.~

### CODE REVIEW

**Comment by garrettwrong** [comment]
PyFFTW in config was restored.  Bugs calling pyfftw should be resolved.

File: src/aspire/numeric/pyfftw_fft.py [hunk 1]
```diff
@@ -129,9 +131,15 @@ def fftn(self, a, axes=None, workers=-1):
...
     def ifftn(self, a, axes=None, workers=-1):
-        mutex.acquire()
 
-        comp_type = a.dtype
+        axes = axes or tuple(range(a.ndim))
+
+        # FFTW_BACKWARD requires complex input array, cast as needed.
+        # See https://pyfftw.readthedocs.io/en/latest/source/pyfftw/pyfftw.html#scheme-table
+        comp_type = complex_type(a.dtype)
+        a = a.astype(comp_type, copy=False)
```
Comment (chris-langfield) [inline src/aspire/numeric/pyfftw_fft.py:140]: Do we want to modify the input array?

File: src/aspire/reconstruction/kernel.py [hunk 2]
```diff
@@ -130,13 +131,15 @@ def convolve_volume_matrix(self, x):
         # Order is important here.  It's about 20% faster to run from 1 through 6 compared with 6 through 1.
         # TODO: Experiment with scipy order; try overwrite_x argument
```
Comment (chris-langfield) [inline src/aspire/reconstruction/kernel.py]: Vestigial comment?

File: src/aspire/reconstruction/kernel.py [hunk 3]
```diff
@@ -130,13 +131,15 @@ def convolve_volume_matrix(self, x):
...
         for i in range(6):
-            x = fft(x, N_ker, i, overwrite_x=True)
+            _pad_width = [(0, 0)] * 6
+            _pad_width[i] = (0, N_ker)
```
Comment (chris-langfield) [inline src/aspire/reconstruction/kernel.py]: Shouldn't this be `N_ker-N`?

**Review by chris-langfield — CHANGES_REQUESTED** [review]

File: src/aspire/reconstruction/kernel.py (hunk 3 above)
Comment (garrettwrong) [reply src/aspire/reconstruction/kernel.py]: Yes, you are right, I dropped it messing around. Thanks! Fixing.

File: src/aspire/reconstruction/kernel.py (hunk 2 above)
Comment (garrettwrong) [reply src/aspire/reconstruction/kernel.py]: Only the second half.... removed.

File: src/aspire/numeric/pyfftw_fft.py (hunk 1 above)
Comment (garrettwrong) [reply src/aspire/numeric/pyfftw_fft.py:140]: In the existing code, we can, but you are totally right, that is dangerous. I'll patch this. Thanks!

Comment (garrettwrong) [reply src/aspire/numeric/pyfftw_fft.py:140]: Wait, nm, I don't think this case occurs.  Are you sure?

Comment (chris-langfield) [reply src/aspire/numeric/pyfftw_fft.py:140]: You mean we always input a complex type array? It's not the worst thing if the array I put in ends up complex but with the same data. Was just a thought

Comment (garrettwrong) [reply src/aspire/numeric/pyfftw_fft.py:140]: No, that isn't what I mean.

What is being modified?

Comment (chris-langfield) [reply src/aspire/numeric/pyfftw_fft.py:140]: if `a.dtype` is a real dtype

```
        comp_type = complex_type(a.dtype)
        a = a.astype(comp_type, copy=False)
```

Comment (chris-langfield) [reply src/aspire/numeric/pyfftw_fft.py:140]: is `a` not having its dtype changed?

Comment (garrettwrong) [reply src/aspire/numeric/pyfftw_fft.py:140]: which a? the one on the right or left of the equals?

Comment (chris-langfield) [reply src/aspire/numeric/pyfftw_fft.py:140]: Oh I see, you're right. Just a relabeling. Nvm

**Review by chris-langfield — APPROVED** [review]

**Review by j-c-c — APPROVED** [review]
This looks good to me.

**Comment by garrettwrong** [comment]
Great, thanks. Merging

**Pull request merged**